src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -1096,7 +1096,7 @@ SchemaOwnerName(Oid objectId)
 static bool
 HasMetadataWorkers(void)
 {
-	List *workerNodeList = ActivePrimaryNodeList();
+	List *workerNodeList = ActivePrimaryNodeList(ShareLock);
```

Comment:
May be best to leave as `NoLock` for now, leave issues here for #2928. `start_metadata_sync_to_node` doesn't appear to currently acquire necessary locks